src/pybindings/pycontroller/BUILD.gn

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import("//build_overrides/build.gni")
+import("//build_overrides/chip.gni")
+import("//build_overrides/pigweed.gni")
+
+import("$dir_pw_build/python.gni")
+
+import("${chip_root}/build/chip/tools.gni")
+import("${chip_root}/src/platform/device.gni")
+import("${dir_pw_unit_test}/test.gni")
+
+if (current_os == "mac") {
+  import("${build_root}/config/mac/mac_sdk.gni")
+}
+
+config("controller_wno_deprecate") {
+  cflags = [
+    "-Wno-deprecated-declarations",
+    "-Wno-shadow",
+    "-Wno-unused-result",
+    "-Wsign-compare",
+    "-Wunreachable-code",
+    "-Wno-macro-redefined",
+  ]
+}
+
+shared_library("CHIPController") {
+  configs -= [ "//build/config/compiler:exceptions_default" ]
+
+  output_name = "PyChip"
+  output_dir = "${target_out_dir}/pybindings"
+  include_dirs = [ "${chip_root}/third_party/pybind11/include" ]
+  if (current_os == "mac") {
+    include_dirs +=
+        [ "${chip_root}/.environment/cipd/python/include/python3.8" ]
+  } else if (current_os == "linux") {
+    include_dirs += [ "/usr/include/python3.9" ]
+  } else {
+    assert(false, "OS not supprted.")
+  }
+
+  sources = [
+    "${root_gen_dir}/include/pybindings/pycontroller/CHIPErrorToExceptionBindings.cpp",
+    "${root_gen_dir}/include/pybindings/pycontroller/CHIPErrorToExceptionBindings.h",
+    "ControllerBindings/PyChip_BLEManager.cpp",
+    "ControllerBindings/PyChip_ControllerDeviceCommissioner.cpp",
+    "ControllerBindings/PyChip_ControllerInitParams.cpp",
+    "ControllerBindings/PyChip_DeviceAddressUpdateDelegate.cpp",
+    "ControllerBindings/PyChip_Inet_IPAddress.cpp",
+    "ControllerBindings/PyChip_Logging.cpp",
+    "ControllerBindings/PyChip_MDNS.cpp",
+    "ControllerBindings/PyChip_Main.cpp",
+    "ControllerBindings/PyChip_Memory.cpp",
+    "ControllerBindings/PyChip_PlatFormManager.cpp",
+    "ControllerBindings/PyChip_Rendezvous.cpp",
+    "ControllerBindings/PyChip_SetupPayload.cpp",
+    "ControllerBindings/PyChip_TransportType.cpp",
+    "clusters/CHIPClusters.cpp",
+  ]
+
+  if (chip_enable_ble) {
+    if (current_os == "mac") {
+      cflags = [
+        "-fobjc-arc",
+        "-fno-common",
+        "-dynamic",
+        "-DNDEBUG",
+        "-g",
+        "-fwrapv",
+        "-O3",
+        "-arch",
+        "x86_64",
+        "-g",
```

Comment:
Do we really need this?  This seems to be duplicating a bunch of default settings... if we just want to _add_ some cflags, should `+=`, not `=`, right?

It's also really odd to only do this if `chip_enable_ble`.

Reply:
Please remove this.